odoo/addons/base/models/ir_actions_report.py

```diff
@@ -420,36 +425,29 @@ def _run_wkhtmltopdf(
             specific_paperformat_args=specific_paperformat_args,
             set_viewport_size=set_viewport_size)
 
-        files_command_args = []
+        def _make_tmp_file(temporary_files, file_suffix, file_prefix, content=None):
+            file_fd, file_path = tempfile.mkstemp(suffix=file_suffix, prefix=file_prefix)
+            with closing(os.fdopen(file_fd, 'wb')) as file:
+                if content:
+                    file.write(content.encode())
+            temporary_files.append(file_path)
```

Comment:
```suggestion
            with tempfile.NamedTemporaryFile(suffix=file_suffix, prefix=file_prefix, delete=False) as f:
                if content:
                    f.write(content.encode())
            temporary_files.append(f.name)
```

Reply:
Although this seems less efficient than the existing code in the sense that it's going to create and use a bunch of empty files if there's no content to write no?

Reply:
No, the only case where there will be no content is when creating the PDF file (for the end result). In that case, it is expected to have an empty file to inject the PDF content afterwards (and retrieve the path of the temporary file for `wkhtmltopdf`).

Reply:
> No, the only case where there will be no content is when creating the PDF file (for the end result). In that case, it is expected to have an empty file to inject the PDF content afterwards (and retrieve the path of the temporary file for `wkhtmltopdf`).

Then why even bother wth the conditional check?

Reply:
Because of this: https://github.com/odoo/odoo/blob/078543587af6d5ef4419ab976fcb2f41de4de91b/odoo/addons/base/models/ir_actions_report.py#L447